### DISCUSSION TITLE
Building Fabric View + Turbo Module

### DISCUSSION THREAD

**Discussion by ian-wd** [callstack/react-native-builder-bob#726] [Q&A] [answered]
Is it possible atm to build a Fabric view along with a turbo module?

I was trying to get it working like how [rn-skia](https://github.com/Shopify/react-native-skia/blob/main/packages/skia/src/specs/NativeSkiaModule.ts) does it but I can't get it working via builder-bob

here's a video showing my problem:
https://github.com/user-attachments/assets/499e92ed-7fab-4c37-997b-d0247135b7dc

seems like whenever I add a turbo module spec it doesn't generate the delegate and interface java files

**Comment by satya164** [comment]
Did you update the `codegenConfig` in `package.json` to use `type: "all"`?

**Reply by ian-wd** [reply]
yeah its using `all` for the type already
tried turning off `includesGeneratedCode` and it does generate it so it must be related to this flag only unsure if this is a codegen related issue or bob-builder one

**Comment by ian-wd** [comment]
@satya164 I think I may have found the cause of this and this may need a fix
- `react-native-builder-bob` only coverts `com.facebook.fbreact.specs`
- while that is true when working with only turbo module or fabric view, when we do fabric + turbo module there's another folder react native codegen creates on android w/c is `com.facebook.react.viewmanagers`
- this doesn't get included on the generated folder hence why I wasn't able to get a sample working with Turbo+Fabric 

here's a sample of generated files:
![image](https://github.com/user-attachments/assets/8e6672e3-1baa-450a-899f-e95be8df3560)

[patchCodegenAndroidPackage](https://github.com/callstack/react-native-builder-bob/blob/main/packages/react-native-builder-bob/src/utils/patchCodegenAndroidPackage.ts) only copies specs from `com.facebook.fbreact.specs` and `com.facebook.react.viewmanagers` are left and not copied over

**Reply by satya164** [reply]
Thanks for debugging @ian-wd 

I'll create an issue to keep track of this

**Reply by satya164** [reply]
https://github.com/callstack/react-native-builder-bob/issues/728

**Comment by atlj** [comment, marked as answer]
@ian-wd thank you so much for the great work! Excellent effort. https://github.com/callstack/react-native-builder-bob/pull/736 Should fix this behavior
